PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,433 @@
+# * University of Minnesota
+# * Oxford University
+#
+# ## Author(s)
+#
+# * Mikhail Milchenko, Department of Radiology, Washington University in St. Louis
+# * Matthew F. Glasser, Department of Anatomy and Neurobiology, Washington University in St. Louis
+#
+# ## Product
+#
+# [Human Connectome Project][HCP] (HCP) Longitudinal Pipelines
+#
+# ## License
+#
+# See the [LICENSE](https://github.com/Washington-University/Pipelines/blob/master/LICENSE.md) file
+
+
+function create_zero_warp
+{
+    local target_image=$1 warp_file=$2
+
+    # Step 1: Extract dimensions from the target image
+    local dim1=$(fslval $target_image dim1)
+    local dim2=$(fslval $target_image dim2)
+    local dim3=$(fslval $target_image dim3)
+    local pixdim1=$(fslval $target_image pixdim1)
+    local pixdim2=$(fslval $target_image pixdim2)
+    local pixdim3=$(fslval $target_image pixdim3)
+
+    local temp=`mktemp volXXX`.nii.gz
+
+    # Step 2: Create a zero-filled image with the same dimensions
+    fslmaths $target_image -mul 0 $temp
+
+    # Step 3: Merge it into a 4D file with three volumes (all zeros)
+    fslmerge -t $warp_file $temp $temp $temp
+
+    # Optional: Set the voxel dimensions correctly if needed
+    fslcreatehd $dim1 $dim2 $dim3 3 $pixdim1 $pixdim2 $pixdim3 1 0 0 0 16 $warp_file
+}
+
+
+set -eu
+
+# Requirements for this script
+#  installed versions of: FSL
+#  environment: HCPPIPEDIR, FSLDIR
+
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/processingmodecheck.shlib" "$@" # Check processing mode requirements
+
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var FSLDIR
+log_Check_Env_Var HCPPIPEDIR_Global
+
+
+log_Msg "Platform Information Follows: "
+uname -a
+"$HCPPIPEDIR"/show_version
+
+#this function gets called by opts_ParseArguments when --help is specified
+function usage()
+{
+    #header text
+    echo "
+$log_ToolName: takes cross-sectional pre-Freesurfer output, as well as longitudinal FreeSurfer output folders.
+Creates volumes and transforms analogous to those produced by preFreesurferPipeline, but in longitudinal template space.
+
+Usage: $log_ToolName PARAMETER...
+
+PARAMETERs are [ ] = optional; < > = user supplied value
+"
+    #automatic argument descriptions
+    opts_ShowArguments
+    
+    #do not use exit, the parsing code takes care of it
+}
+
+#arguments to opts_Add*: switch, variable to set, name for inside of <> in help text, description, [default value if AddOptional], [compatibility flag, ...]
+#help info for option gets printed like "--foo=<$3> - $4"
+
+#TSC:should --path or --study-folder be the flag displayed by the usage?
+opts_AddMandatory '--path' 'StudyFolder' 'path' "folder containing all timepoins and templates" "--path"
+#The following options
+opts_AddMandatory '--subject'   'Subject'   'subject ID' "Subject label. Note: this is distinct subject label. Use separate labels for template and timepoints"
```

Comment:
So `--subject` here does not have the same meaning as in the other structural pipelines?  If so, that is going to be confusing, and we should probably come up with a different argument name that reflects the true "subject ID" (rather than the session ID).

Reply:
This is subject label and is uniform across structural pipelines.

Reply:
I don't think we have used the term "label" for this, and the description didn't really tell me what was supposed to happen.  It should probably say something like `to be used in the output folder name as '<Subject>.long.<Template>'`.  I'm still not clear on exactly what is expected, though, so it should probably also have instructions like "should not include the visit specifier" or "should indicate what timepoints are included when run in template mode" or something.

But yes, if it isn't asking for the subject folder name (HCA1234567_V1_MR), we should probably give it a different flag than what we have previously used to specify the subject folder in other pipelines.

Reply:
I'm only tangentially able to follow all the stuff that is going on, but at some point I suggested that we take this opportunity to fix the confusion between `subject` and `session` in the various scripts, by introducing new descriptions and argument names (but preserving the existing argument names as backward compatible options).  QuNex made this switch in their terminology a while ago, which was a helpful change.  Is this something that is still under consideration?

Reply:
1. I Updated option description in question to refer to 'subject ID' with example of how it's used.
2. Currently --session replaces --subject in all structural pipelines, with legacy switch --subject still available. 
'subjectID' may still be confusing but probably better as some of my code takes both --session (in the sense of input folder) and --subjectID as input. 
If agreed I'll make this change.

Reply:
I'm not following.  The code snippet under review still uses `--subject` as the main variable name.

Reply:
So, the problem we face is that `--subject` needs to be preserved as a legacy switch for what is now going to be better coded as the `--session` flag, but for the purposes of the longitudinal pipeline we now need a flag for the "true" subject, for which you are using (or planning to use) `--subjectID`?